src/main/java/com/kainos/ea/daos/RoleDao.java

```diff
@@ -15,44 +15,60 @@
 import java.util.List;
 
 public class RoleDao {
-    /**
-     * Gets.
-     * @return Returns list of open job roles
-     * @throws SQLException
-     * @throws FailedToRetrieveException
-     */
+
     public List<RoleResponse> getAllJobRoles()
             throws SQLException, FailedToRetrieveException {
+        return getOrderedJobRoles(null, null);
+    }
+
+    public List<RoleResponse> getOrderedJobRoles(
+            final String orderBy, final String direction)
+            throws SQLException, FailedToRetrieveException {
         List<RoleResponse> roles = new ArrayList<>();
 
+        // Validate orderBy and direction to prevent SQL injection
+        List<String> validColumns = List.of(
+                "roleName", "location", "band",
+                "capability", "closingDate", "status");
+        if (orderBy != null && direction != null
+                && (!validColumns.contains(orderBy)
+                || (!"ASC".equalsIgnoreCase(direction)
```

Comment:
And these as well?

Reply:
don't see an enum for this

Reply:
this is not resolved, please don't mark comments as resolved. typically that's done by the reviewer once they look at the code again.